### PR TITLE
feat(cli): add doctor support bundle export

### LINKED ISSUE #232: feat(cli): add doctor support bundle export for shareable diagnostics
## Background

When first-run setup or live orchestration fails, users currently have to collect runtime evidence by hand across multiple locations: `WORKFLOW.md`, managed project config, `status.json`, recent run records, `events.ndjson`, `worker.log`, and `orchestrator.log`.

That manual collection cost is small for maintainers who know the runtime layout, but high for new adopters and outside contributors. In an open-source project, this directly slows issue reporting and weakens product trust because the first support interaction starts with "please gather these files" instead of a reproducible artifact.

A built-in diagnostics bundle is a high-leverage usability improvement: it does not change orchestration semantics, but it makes bugs easier to report, easier to triage, and easier to fix.

## Proposed Changes

1. Extend `gh-symphony doctor` with a bundle export mode such as `gh-symphony doctor --bundle [path]`.
2. Reuse the existing doctor checks and runtime filesystem readers to collect a bounded, shareable support bundle for the active managed project (or an explicit `--project-id`).
3. Include at least these artifacts:
   - machine-readable `doctor --json` output
   - active managed project config (`project.json`) and binding metadata
   - repository `WORKFLOW.md`
   - runtime `status.json` and `issues.json`
   - recent run metadata (`run.json`) for the latest active/recent runs
   - bounded tails of `events.ndjson`, `worker.log`, and `orchestrator.log`
4. Redact obvious secret-bearing values before writing the bundle, including GitHub tokens, API keys, and credential-like env assignments.
5. Print a concise summary of what was collected, what was redacted, and the final output path so the bundle can be attached directly to a GitHub issue.
6. Keep the feature safe by default: bounded log tails only, no recursive workspace copy, and explicit opt-in if full logs are ever supported later.

## Affected Files

| File | Change | Scope |
| --- | --- | --- |
| `packages/cli/src/index.ts` | Register the new `doctor --bundle` option and help text | CLI surface |
| `packages/cli/src/commands/doctor.ts` | Plan bundle contents, run diagnostics, and write the export artifact | Main feature |
| `packages/cli/src/support/bundle.ts` | New shared helper for redaction, bounded file capture, and archive layout | Support tooling |
| `packages/cli/src/commands/logs.ts` | Reuse or extract bounded-tail helpers for run/event log capture | Shared log reading |
| `packages/cli/src/commands/doctor.test.ts` | Cover bundle creation, redaction, missing-file tolerance, and explicit project selection | Verification |
| `README.md` | Document the support-bundle workflow for bug reports | Adoption |
| `packages/cli/README.md` | Add command examples and bundle contents summary | Adoption |

## Verification

- `pnpm --filter @gh-symphony/cli test`
- `pnpm --filter @gh-symphony/cli typecheck`
- `gh-symphony doctor --bundle`
- `gh-symphony doctor --bundle ./tmp/support-bundle`

## References

- `packages/cli/src/commands/doctor.ts`
- `packages/cli/src/commands/logs.ts`
- `packages/dashboard/src/store.ts`
- `packages/core/src/observability/fs-reader.ts`

### PR DESCRIPTION
**Pull request by moncher-dev**
## Issues

- Fixes #232

## Summary

- Adds `gh-symphony doctor --bundle [path]` to export a redacted diagnostics bundle for the selected managed project.
- Writes a deterministic bundle layout with `manifest.json`, `doctor.json`, config/runtime artifacts, recent run metadata, and bounded log/event tails.
- Hardens the review feedback paths: quoted JSON-style raw text secrets are redacted, oversized single-line tails remain non-empty, malformed optional JSON artifacts are recorded as missing, and shared observability redaction stays key-focused unless support bundle export opts into deep text scanning.

## Changes

- Added `packages/cli/src/support/bundle.ts` for safe allowlisted bundle writes, bounded tails, manifest accounting, missing-artifact tracking, and recent run selection.
- Wired `--bundle`, `--bundle --json`, `--bundle --project-id`, and `--bundle --fix` rejection into the doctor command and CLI help.
- Added support-bundle-specific diagnostics redaction so bundle JSON values and raw text are scrubbed without changing orchestrator persistence semantics.
- Added regression coverage for quoted raw secrets, large single-line log tails, malformed config/project JSON tolerance, explicit/default paths, explicit project selection, truncation recording, and fixture secret absence.
- Documented support bundle usage in `README.md` and `packages/cli/README.md`.

## Evidence

- `pnpm --filter @gh-symphony/cli test -- doctor.test.ts`
- `pnpm --filter @gh-symphony/core test -- redaction.test.ts`
- `pnpm --filter @gh-symphony/orchestrator test -- service.test.ts`
- `pnpm --filter @gh-symphony/cli typecheck`
- `pnpm --filter @gh-symphony/core typecheck`
- `git diff --check`
- `pnpm lint && pnpm test && pnpm typecheck && pnpm build`
- Manual check from the initial implementation remains applicable: built CLI ran `doctor --bundle` and `doctor --bundle ./tmp/support-bundle` in a temporary initialized runtime fixture; generated bundles were checked with `grep -R "abc123\|sk-xxx" ...` and no fixture secret values were found.

## Human Validation

- [ ] Confirm `gh-symphony doctor --bundle` creates a usable bundle in an initialized repository runtime.
- [ ] Confirm the bundle contents are safe to attach to a public issue after reviewer inspection.
- [ ] Confirm `--json` summary shape works for support automation.

## Risks

- Redaction remains pattern-based for obvious secret-bearing values; reviewers should inspect whether additional project-specific secret formats need follow-up coverage.
- The 5 MiB bundle byte cap is treated as a recorded soft target in `manifest.bundleBytes`; per-artifact run/log limits still provide the hard MVP bounds.

### CODE REVIEW

File: packages/core/src/observability/redaction.ts [hunk 1]
```diff
@@ -7,38 +7,229 @@ const SENSITIVE_KEY_SUBSTRINGS = [
...
+  redacted = replaceAndCount(
+    redacted,
+    /^([A-Z0-9_]*(?:SECRET)\w*\s*=\s*)([^\s]+)/gim,
+    "secret_key",
+    counts,
+    "$1[REDACTED]"
+  );
+  redacted = replaceAndCount(
+    redacted,
+    /\b(token\s*:\s*)([^\s,}\]]+)/gi,
```
Comment (hojinzs) [inline packages/core/src/observability/redaction.ts]: **[P1 — security] Confirms Codex finding: JSON-quoted / prefixless secrets in raw text are not redacted.**

I reproduced this independently. The `token:` / `secret:` / `apiKey:` patterns require an unquoted `key: value` shape, so a JSON line leaks any value lacking a known prefix:

```
input : {"event":"worker","token":"s3cr3tValueNoPrefix","apiKey":"k3yNoPrefix"}
output: {"event":"worker","token":"s3cr3tValueNoPrefix","apiKey":"k3yNoPrefix"}  // unchanged
```

This matters because the bundle captures `events.ndjson` (JSON-per-line) and raw `worker.log` / `orchestrator.log` tails. `fs-store` pre-redacts orchestrator events by object key, so NDJSON has defense-in-depth — but `worker.log`/`orchestrator.log` are raw and unredacted at source, and the issue explicitly requires redacting "NDJSON line content" and raw logs as a **security** requirement.

Suggested fix: extend the key/value text patterns to also match quoted JSON forms, e.g.
`/("?\b(?:token|secret|apiKey|api[_-]?key)"?\s*[:=]\s*)"?([^\s,"}\]]+)"?/gi`
and add a fixture line `{"token":"abc123","apiKey":"k3y"}` to the redaction test asserting the values are absent.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/cli/src/support/bundle.ts [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+      await handle.read(buffer, 0, length, start);
+    }
+
+    let text = buffer.toString("utf8");
+    let truncated = start > 0;
+    const reasons: string[] = [];
+    if (start > 0) {
+      reasons.push("maxLogBytes");
+      const firstNewline = text.indexOf("\n");
+      text = firstNewline >= 0 ? text.slice(firstNewline + 1) : "";
```
Comment (hojinzs) [inline packages/cli/src/support/bundle.ts]: **[P2 — correctness] Confirms Codex finding: a >64 KiB single-line log produces an empty `.tail`.**

Reproduced with a 70 KiB newline-free file using this exact algorithm: `start=6144`, captured tail length = **0**. When `start > 0` and there is no `\n` in the byte window, `text` is set to `""`, so the artifact is empty — losing all recent diagnostics for precisely the large-log case the bundle exists to capture (long single-line JSON logs, big stack traces, base64 blobs in `worker.log`/`orchestrator.log`).

Suggested fix: when no newline is found, keep the captured bytes instead of discarding them, and record the partial-line truncation:

```ts
if (start > 0) {
  reasons.push("maxLogBytes");
  const firstNewline = text.indexOf("\n");
  if (firstNewline >= 0) {
    text = text.slice(firstNewline + 1);
  }
  // else: no newline in window — keep the (partial) tail rather than emptying it
}
```

Please add a test for a single-line log larger than `maxLogBytes` asserting the tail is non-empty and truncation is recorded.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/core/src/observability/redaction.ts [hunk 3]
```diff
@@ -7,38 +7,229 @@ const SENSITIVE_KEY_SUBSTRINGS = [
...
+  value: unknown,
+  counts: Map<RedactionClass, number>
+): unknown {
   if (Array.isArray(value)) {
-    return value.map((item) => redactValue(item));
+    return value.map((item) => redactValue(item, counts));
+  }
+
+  if (typeof value === "string") {
+    return redactTextValue(value, counts);
```
Comment (hojinzs) [inline packages/core/src/observability/redaction.ts:74]: **[P2 — scope/perf risk] This silently changes a shared orchestrator hot path.**

Recursing `redactTextValue` into every string makes `redactObservabilitySecrets` (key-only until now) run all 11 regexes over every string field of every object. That function is called by `packages/orchestrator/src/fs-store.ts:177` (every run snapshot) and `:182` (every event append) — a high-frequency persistence path. Two consequences:

1. **Performance**: 11 regex passes per string per persisted event/run, on the orchestrator's hot path.
2. **Over-redaction / observability loss**: legitimate run/event strings containing `token:`, `secret:`, `sk-…`, `ghp_…` substrings now become `[REDACTED]` in persisted observability data. The PR's own workpad notes you already had to "narrow redaction so `workspaceKey` remains unredacted" for `service.test.ts` — direct evidence of false positives leaking into unrelated code.

A CLI support-bundle feature shouldn't change the orchestrator's persisted observability semantics. Recommend keeping `redactObservabilitySecrets` key-only and having the bundle apply the text-value redaction explicitly (e.g. a separate `deepRedactText` the bundle opts into), so the blast radius stays in the support tooling.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/cli/src/support/bundle.ts [hunk 4]
```diff
@@ -0,0 +1,507 @@
...
+  } catch (error) {
+    recordMissing(state, destinationPath, sourcePath, error);
+    return;
+  }
+
+  try {
+    await writeJsonArtifact(state, destinationPath, JSON.parse(raw));
+  } catch (error) {
+    throw new Error(
+      `Failed to redact/write JSON artifact ${sourcePath}: ${formatError(error)}`
```
Comment (hojinzs) [inline packages/cli/src/support/bundle.ts:259]: **[P2 — robustness] A malformed `config.json`/`project.json` hard-fails the whole bundle.**

A read error is tolerated (recorded as missing, line 234), but `JSON.parse(raw)` throwing here rejects the entire `createSupportBundle`. A corrupt config is exactly the kind of failure a user wants to file a support bundle for — and `listRecentRuns` already tolerates a bad `run.json` (try/catch at line 437), so the behavior is inconsistent across artifacts.

The issue scopes hard failures to "redaction/write failure / invalid project selection / unwritable output / internal invariant violation" — a malformed *input* artifact isn't one of those. Recommend recording the parse error in `manifest.missing` (with the reason) and continuing, matching the run.json behavior.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/cli/src/support/bundle.ts [hunk 5]
```diff
@@ -0,0 +1,507 @@
...
+  configFilePath,
+  orchestratorLogPath,
+  projectConfigPath,
+} from "../config.js";
+
+export const SUPPORT_BUNDLE_LIMITS = {
+  maxRuns: 3,
+  maxLogBytes: 64 * 1024,
+  maxLogLines: 500,
+  maxBundleBytes: 5 * 1024 * 1024,
```
Comment (hojinzs) [inline packages/cli/src/support/bundle.ts:27]: **[Minor] `maxBundleBytes` is recorded but never enforced.**

It's written into `manifest.limits` but no code accumulates total bundle size or acts on this value. Per-artifact bounds (3 runs, 64 KiB tails) keep the total small in practice, so this is minor — but surfacing an unenforced "limit" in the manifest is misleading to anyone inspecting it. Either accumulate written bytes and enforce a soft cap (with accounting in `manifest.truncations`), or rename/document it as an advisory target.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by hojinzs — CHANGES_REQUESTED** [review]
## PR #340 완성도 검증 (Round 1/6)

원본 이슈 #232 대비 요구사항/수용조건 검증 + Smoke Test + 코드 리뷰를 수행했습니다.

### 1. 요구사항 / 수용 조건 커버리지 — ✅ 사실상 충족
이슈 #232의 Acceptance Criteria를 코드/테스트로 대조한 결과 모두 구현되어 있습니다: `--bundle` / `--bundle <path>` / `--project-id` 선택, `manifest.json`+`doctor.json`+redacted config+`WORKFLOW.md`+runtime status/issues+bounded run·log tail 레이아웃, `manifest.missing` 기록(미존재 시 비실패), 구조적·텍스트 redaction, 최소 시크릿 픽스처 부재 단언, truncation 기록, 비재귀 allowlist 수집, `--json` 요약, `--fix` 조합 거부. 문서(README/패키지 README)도 반영됨.

### 2. Smoke Test 결과 (Preview 환경 없음 → 로컬 빌드/유닛)
- `pnpm -r build`: ✅ 성공
- `pnpm --filter @gh-symphony/cli typecheck` / `@gh-symphony/core typecheck`: ✅ 통과
- 번들 테스트 4종(explicit path / default timestamped / project 선택 / `--fix` 거부) + `redaction.test.ts` + `index.test.ts` + `help.test.ts`: ✅ 전부 통과
- 빌드된 CLI 수동 확인: `doctor --fix --bundle` → `Option '--fix' cannot be used with '--bundle'` (이슈 명세 문구 일치), `doctor --help`에 `--bundle [path]` 노출 ✅
- ⚠️ 참고: `doctor.test.ts`의 인증 테스트 2건(`accepts env-token auth…`, `falls back to gh auth…`)이 실패하나, 이는 **본 PR과 무관**합니다. 해당 테스트는 Node 버전을 목킹하지 않아 샌드박스의 Node v22(프로젝트 요구 24+)에서 `report.ok===false`가 됩니다. CI의 `Test`/`Container Smoke` 체크는 Node 24에서 success. 라이브 GitHub 프로젝트가 없어 end-to-end `doctor --bundle` 실런은 불가하나, 모킹된 프로젝트 선택을 사용하는 유닛 테스트가 전체 번들 생성 경로를 커버합니다.

### 3. 코드 리뷰 — 변경 요청 (인라인 코멘트 5건)
구조와 테스트 커버리지는 견고하나(번들 91.6%/redaction 97%), 머지 전 처리할 실질 결함이 있습니다. Codex의 P1/P2 두 건을 **독립적으로 재현하여 사실 확인**했습니다:
- **P1 (보안)**: JSON 따옴표/접두사 없는 시크릿이 raw text(`events.ndjson`, `worker.log`, `orchestrator.log`)에서 redaction되지 않음 — 재현 확인. 이슈가 명시한 보안 요구사항.
- **P2 (정확성)**: 64 KiB 초과 단일 라인 로그 → `.tail` 빈 파일(진단 전손) — 70 KiB 파일로 재현 확인.
- **P2 (범위/성능)**: `redactObservabilitySecrets`가 키 전용 → 모든 문자열 텍스트 redaction으로 확장되어 오케스트레이터 `fs-store` 핫패스(run/event 영속화)의 동작을 무단 변경(성능·과다 redaction; workpad의 `workspaceKey` 좁히기 이력이 방증).
- **P2 (견고성)**: 잘못된 `config.json`/`project.json` 파싱 실패 시 번들 전체 하드 실패 — 정작 지원이 가장 필요한 상황. `run.json` 처리와 불일치.
- **Minor**: `maxBundleBytes`가 manifest에 기록되나 어디서도 미적용.

### 결론
요구사항은 거의 충족하고 폐기할 PR이 아닙니다(기능 견고, ~90% 완성). 다만 이슈가 명시한 **보안 요구사항(P1)** 과 **정확성 결함(P2)** 이 실재하므로 **Request Changes**. P1/P2와 fs-store 범위 확장 처리 후 재검토하면 Approve 가능한 상태로 판단합니다. (핑퐁 1/6)

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/core/src/observability/redaction.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (moncher-dev) [reply packages/core/src/observability/redaction.ts]: Fixed in ac08e70. Raw text redaction now matches quoted JSON-style `token`, `secret`, and `apiKey` fields, and the regression fixture asserts the quoted secret values are absent from generated bundles.

File: packages/cli/src/support/bundle.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (moncher-dev) [reply packages/cli/src/support/bundle.ts]: Fixed in ac08e70. The bounded-tail reader now keeps the captured byte window when no newline exists instead of writing an empty artifact, records `partialLine` in the truncation reason, and has a regression test for an oversized single-line `worker.log`.

File: packages/core/src/observability/redaction.ts (hunk 1 above)
Comment (moncher-dev) [reply packages/core/src/observability/redaction.ts]: Fixed in ac08e70. Added quoted JSON-style key/value matching for raw text redaction and fixture coverage for `{"token":...,"secret":...,"apiKey":...}` values in NDJSON/log content. The bundle-level assertion now verifies those values do not appear anywhere in the generated bundle.

File: packages/cli/src/support/bundle.ts (hunk 2 above)
Comment (moncher-dev) [reply packages/cli/src/support/bundle.ts]: Fixed in ac08e70. If the 64 KiB tail window has no newline, the support bundle now preserves the partial tail instead of replacing it with an empty string. Added a test that writes an oversized single-line log, checks the tail is non-empty and contains the final marker, and verifies truncation records `partialLine`.

File: packages/core/src/observability/redaction.ts (hunk 3 above)
Comment (moncher-dev) [reply packages/core/src/observability/redaction.ts:74]: Fixed in ac08e70. `redactObservabilitySecrets` and `redactObservabilitySecretsWithStats` are back to key-focused structured redaction, preserving the orchestrator persistence hot path semantics. Support bundle export now opts into the deeper string-value scan through `redactObservabilityDiagnosticsWithStats`, with tests covering both behaviors.

File: packages/cli/src/support/bundle.ts (hunk 4 above)
Comment (moncher-dev) [reply packages/cli/src/support/bundle.ts:259]: Fixed in ac08e70. Malformed optional JSON input artifacts are now recorded via `manifest.missing` and bundle creation continues, matching the existing run metadata tolerance. Added coverage for malformed `config.json` and `project.json`.

File: packages/cli/src/support/bundle.ts (hunk 5 above)
Comment (moncher-dev) [reply packages/cli/src/support/bundle.ts:27]: Addressed in ac08e70 by making the soft target explicit in the manifest: bundle writes are byte-accounted and `manifest.bundleBytes` records `written`, `softMax`, and whether the target was exceeded. The hard MVP bounds remain the allowlisted files, max 3 runs, and bounded log/event tails.

File: packages/core/src/observability/redaction.ts [hunk 6]
```diff
@@ -7,38 +7,242 @@ const SENSITIVE_KEY_SUBSTRINGS = [
...
+  redacted = replaceAndCount(
+    redacted,
+    /((?:"secret"|'secret'|secret)\s*:\s*)(?:"([^"]*)"|'([^']*)'|([^\s,}\]]+))/gi,
+    "secret_key",
+    counts,
+    '$1"[REDACTED]"'
+  );
+  redacted = replaceAndCount(
+    redacted,
+    /((?:"apiKey"|'apiKey'|apiKey)\s*:\s*)(?:"([^"]*)"|'([^']*)'|([^\s,}\]]+))/g,
```
Comment (hojinzs) [inline packages/core/src/observability/redaction.ts:181]: **[Nit — non-blocking, follow-up only]** Minor inconsistency: the `token` (L167) and `secret` (L174) raw-text patterns are case-insensitive (`/gi`), but this `apiKey` pattern is `/g` (case-sensitive). So a raw NDJSON/log line like `{"apikey":"..."}` or `{"APIKEY":"..."}` would not be scrubbed by the text path (the structured-key path *is* case-insensitive, but `events.ndjson`/`worker.log` are processed as raw text).

This is **not a blocker**: the issue's required fixture (`apiKey: xxx`, camelCase) is covered and passes, and the PR's "Risks" section already scopes pattern-based redaction hardening as follow-up. Flagging only so a future redaction-coverage pass aligns this to `/gi` for symmetry with the other two patterns. No change required to merge.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by hojinzs — APPROVED** [review]
## PR #340 완성도 검증 — 최종 의견: ✅ Approve (Round 2/6)

Round 1에서 제기한 5건(P1 1 / P2 3 / Minor 1)을 **코드와 회귀 테스트로 직접 재검증**했습니다. 회신 코멘트를 신뢰하지 않고 `ac08e70` 소스를 대조했습니다.

### 1. Round 1 지적사항 해소 검증 — 전부 해결 ✅

| 항목 | 검증 결과 |
| --- | --- |
| **P1 (보안)** 따옴표 JSON `token`/`secret`/`apiKey` 미redaction | `redaction.ts` L165–185 인용/비인용 패턴 추가. `events.ndjson` 픽스처에 `{"token":...,"secret":...,"apiKey":...}` 라인 추가 후 번들 전체에서 `json_*_value` 부재 단언 → **이슈의 번들 레벨 보안 단언 충족** |
| **P2 (정확성)** 64 KiB 초과 단일 라인 → 빈 `.tail` | `bundle.ts` L316–324 newline 부재 시 부분 tail 보존 + `partialLine` 기록. `maxLogBytes+1024` 회귀 테스트가 `tail-marker` 존재·truncation 기록 단언 |
| **P2 (범위/성능)** 오케스트레이터 핫패스 동작 변경 | `redactObservabilitySecrets`(fs-store L177/182) → `redactStringValues:false`(키 전용 복귀). 번들만 `redactObservabilityDiagnosticsWithStats`(deep) opt-in. "키 전용 유지" 회귀 테스트 2건 추가 → 블라스트 반경 지원 도구로 격리 |
| **P2 (견고성)** 잘못된 config/project JSON 하드 실패 | `copyJsonArtifact` L250–261 `SyntaxError` → `manifest.missing` 기록 후 계속(run.json과 일관). 회귀 테스트가 `missingCount>=2` 및 미실패 단언 |
| **Minor** `maxBundleBytes` 미적용 | `manifest.bundleBytes`(`written`/`softMax`/`exceeded`) 바이트 어카운팅으로 advisory 명시화 |

### 2. Smoke Test (Preview 환경 없음 → 로컬/CI)
- `@gh-symphony/core redaction.test.ts`: 4/4 ✅
- `@gh-symphony/cli` 번들 테스트(quoted-secret 부재 / oversized single-line tail / malformed JSON / explicit·default path / project 선택 / `--fix` 거부): ✅ (전체 doctor 37 pass)
- typecheck cli/core/orchestrator: ✅ / CI `Test`·`Container Smoke`(Node 24): success ✅
- ⚠️ 샌드박스 Node v22 한정 인증 테스트 2건 실패는 **본 PR 무관**(Round 1과 동일, Node 24 CI에서 green). 라이브 GitHub 프로젝트 부재로 end-to-end 실런 불가하나 모킹 픽스처가 전체 번들 경로를 커버.

### 3. 코드 리뷰
인라인 nit 1건(`apiKey` raw-text 정규식 `/g`→`/gi` 대칭성, **비차단/후속**)만 남김. 이슈 필수 픽스처는 통과하며 PR "Risks"가 패턴 기반 redaction 후속 강화를 이미 명시함.

### 결론
원본 이슈 #232의 모든 Acceptance Criteria 충족, Round 1 블로커 5건 전부 회귀 테스트와 함께 실해결 확인, CI green. 남은 항목은 이슈 명세 범위 밖의 비차단 nit뿐이므로 추가 핑퐁은 오버 엔지니어링에 해당합니다. **Approve** 합니다. (핑퐁 2/6 종료)

---
_Generated by [Claude Code](https://claude.ai/code)_
